### PR TITLE
913 landing page for ebook

### PR DESCRIPTION
**Pull request by nikeshashar**

### CODE REVIEW

**Comment by nikeshashar** [comment]
Hey @peter-miklos I've made a PR here so we can review together

**Comment by nikeshashar** [comment]
image

Could we put this into a new `container` and see if it looks any better with a white background?

File: source/sass/_rows.scss [hunk 1]
```diff
@@ -136,6 +136,66 @@
...
+    }
+
+    label {
+      text-transform: none;
+      margin-right:15px;
+      margin-left:15px;
+      float: none;
+      display: inline-block;
+      vertical-align: middle;
+    }
```
Comment (nikeshashar) [inline source/sass/_rows.scss]: We should probably make the font larger here. I'm only testing on Chrome at the moment but it looks quite small.

File: source/ebook.html.haml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+- description "Makers Academy is London's leading Web Developer Bootcamp. Learn to code in 12 Weeks by enrolling on one of our programming / web development courses! Learn Ruby, JavaScript and the best software practices."
+%section.hero-splash{style: "background-image: url(#{image_path("backgrounds/dan-and-students-overhead-70.jpg")});"}
+  .container
+    .centered-row
+      %h1 Learn Ruby In 15 Minutes
+      %h4.subheader Download the guide
+      %P Get a free copy of our Learn Ruby In 15 Minutes e-book by signing up to our newsletter. It's perfect for anyone that is interested in taking the plunge by learning to code. It is a good start into the Ruby language if you've never had any experience before and we've made it as easy to understand as possible.
+    = partial :"quiz-question", locals: { segments: "website-lr15" }
+    = partial :"email-capture", locals: { segments: "website-lr15" }
```
Comment (nikeshashar) [inline source/ebook.html.haml]: does the quiz-question partial need a local segment?

Comment (nikeshashar) [inline source/ebook.html.haml]: change the email capture segment to be `website-ebook-lr15`

**Comment by nikeshashar** [comment]
Just a thought, would it be nice to have a green background on the text `You're correct!` when the answer is right just flash up for a bit? I feel visuals + color make for better customer interaction.

**Comment by nikeshashar** [comment]
@alexandergimson we'll need to rework the existing `website-xxxx-lr15` segments to send out a link to the new book. 

+ update the pdf online too!

File: source/ebook.html.haml (hunk 2 above)
Comment (peter-miklos) [inline source/ebook.html.haml]: email capture segment updated

Comment (peter-miklos) [inline source/ebook.html.haml]: I think not. Local segment removed

File: source/sass/_rows.scss (hunk 1 above)
Comment (peter-miklos) [inline source/sass/_rows.scss]: I think it's bit small. Made it larger a bit

**Comment by nikeshashar** [comment]
From Peter : 

Hey Nikesh, hope you’re well. As requested I went back where I was on Wednesday with the unit tests in js. My issue was (and still is) testing with `jasmine-node` and `jquery`. I read plenty of articles and tried many approaches with no real success. I don’t want to summarize now all the issues I had, and I'm not sure I've chosen the right approach. Can you pls help in getting on the right track in testing this jquery plugin (quiz.js.erb)? thanks in advance

**Comment by peter-miklos** [comment]
From Nikesh:
Hey Peter, I’ve posted this across on the ticket. Feedback from Rachel and Dan was to try to Unit test your single function that you made.

Respond:
I’ve already created test cases for testing the main function (called quiz function) with jasmine-node and jsdom. Is this approach acceptable in unit tests?
If yes, then the issue is that the jsdom does not understand the haml code, so the relevant view cannot be loaded/imported, and html equivalent of haml needs to be copy/pasted into the test file, in order to make it work. Could I get any advice on it? (worst case I will use .erb for the quiz part only, then it should be okay) Thank you.

**Comment by peter-miklos** [comment]
Further tests and remaining changes added to the code in the last commit (5835cbb). Details:
- Test cases added to quiz-spec.js;
- Changing response's background color to green if correct values are submitted;
- Response messages updated as requested;
- Styling of ebook page updated;
- quiz.js and quiz-spec files refactored;

**Comment by peter-miklos** [comment]
Can you please check the content of the pushed code and let me know your feedback? 
This version still handle only one quiz question.
I’m already working on a change that can randomly choose a question from a set of quiz questions from the already existing quiz-inputs.yml file.

**Comment by rachelcdavies** [comment]
[Uploading Screen Shot 2017-06-13 at 17.37.51.png...]

**Comment by rachelcdavies** [comment]
@peter-miklos I tried running tests on this branch and got the above errors.

**Comment by peter-miklos** [comment]
@rachelcdavies can you please run npm install before running npm test? I've added jsdom to package.json

**Comment by peter-miklos** [comment]
regarding ruby tests, I'll remove ebook_spec.rb and the added gems (capybara and friends). I moved to unit tests in js, testing the quiz function.

**Comment by rachelcdavies** [comment]
@peter-miklos ah thanks for the reminder to run npm install. Now when I run npm test they all pass :-) I think as you suggest removing ebook_spec.rb and the added gems (capybara and friends) makes sense at this point.

**Comment by peter-miklos** [comment]
The followings have been removed:
- ebook_spec.rb
- capybara related settings from spec_helper
- capybara, capybara-webkit, selenium-webdriver gems

**Comment by nikeshashar** [comment]
image

The question still shows after I've submitted the answer. Is there a way to just have the email box and a better visual for `You're correct`? 

Maybe slightly larger?

**Comment by peter-miklos** [comment]
@nikeshashar Changes have been added and pushed.

**Comment by peter-miklos** [comment]
@nikeshashar 
I’was investigating the possible ways of using more than one quiz question in the quiz (randomly choose one) and preferably from an external source (from a yml file). I found the following options:
1.) Very basic and not nice: add an array of quiz objects into the quiz.js (each quiz has a question, answer(s), labels connected to the answers and an explanation), and randomly choose one and use its content on the page. No external source is used, and js needs to be modified if there is a change in a quiz.
2.) Import a yml file into quiz.js: the only way I found, is to use ruby code in js file (middleman lets you do that if the extension is js.erb) for importing the content of the yml file. I used this solution in the first few versions of my code when I dealt with set of quiz question from an external data source (quiz_inputs.yml). Since we can get the data, we can do what we need to do with them. But, there is an ISSUE with it: if we use ruby code in the jQuery plugin (I know it’s strange, but middleman can handle it), we cannot run test cases. The current JS test framework (jasmine-node with jsdom) does not understand ruby code in the js file, so stops without running any tests. Perhaps there is a workaround for this, but I haven’t found any :(
3.) Use only ruby code for the business logic: I’m in the middle of it, and I have an idea how the process should look like. I think playing with the layout can be a bit more challenging if no or just minimal jQuery is used. I assume it will be easier to test.

I don’t want to overcomplicate it, so I have a few questions:
- Please let me know your view on the above options.
- Can you please send me some more quiz ideas?
- Can we maximise the possible number of input fields for the answer? I mean in the current quiz we use 2 input fields for the answers (one for var a and one for var b). Can we say that the maximum number of input fields in the answer is 3 or 4? 
Thank you.

**Comment by nikeshashar** [comment]
Hey @peter-miklos - we're just discussing the MVP at the moment. Don't worry too much about the multiple quiz questions yet. We don't know if this page will work yet or not.

**Comment by rachelcdavies** [comment]
:shipit:

**Comment by nikeshashar** [comment]
Not quite yet @rachelcdavies @peter-miklos - I've got to check the email segment works and possibly update that too.

**Comment by nikeshashar** [comment]
@peter-miklos @rubenkostucki @rachelcdavies @alexandergimson - this looks like it's done now. Thank you everyone for your input. I'll merge the PR once approved tomorrow!

**Pull request merged**